doc/design/container_logging.md

```diff
@@ -0,0 +1,24 @@
+# Container VM output logging in VIC
+
+When a container VM process produces output, that output is sent down through a serial connection into the vSphere backend where it is written to a logfile. When a user wants to see this output, they use the `docker logs` command, which reads from this logfile and displays the log entries on the command line in order from oldest to newest.
+
+However, in order for users to be able to use the `--since` option with `docker logs`, we must also couple timestamps to these log entries so that the user can selectively filter log messages to be displayed based on when those entries occurred.
+
+### Requirements
+
+The container VM logging mechanism must:
+  1. Wrap each log entry in a JSON struct containing both the log message(s) in that entry, and the timestamp in which that entry occurred. 
```

Comment:
This is not actually a requirement - it's an implementation choice. It _could_ be a requirement if it's talking about what the Docker API should return, but that's distinct from what needs to be persisted by the containerVM.

The requirements here are:
* allow separation of stdout & stderr in the tty case (include this here to inform implementation choice, doesn't necessarily have to be included in impl)
* associate a section of output with a timestamp

Questions:
* should timestamps be recorded for non-tty processes (which may be generating binary data)
* should the timestamps be recorded inline with the output, or separately in a manner such that they can be combined with the output
* what is the separator for output sections
  * when using tty (probably newline)
  * when not using tty (could be arbitrary data)
  * should this be configurable
* is the time when the section was started, or when it was finished
* timestamping is an overhead - is this configurable per container